src/target/cortexm.c

```diff
@@ -296,6 +296,16 @@ bool cortexm_probe(ADIv5_AP_t *ap)
 		target_check_error(t);
 	}
 
+	bool connect_srst = platform_srst_get_val();
+	/* Enable debug access for probing.*/
+	target_halt_request(t);
```

Comment:
Why do you halt a running target on probe?  This probe function should be as unintrusive as possible.
Halting debug will need to be enabled for the DEMCR to take effect, but this should be done inside the if statement below:
```
target_mem_write32(t, CORTEXM_DEMCR, CORTEXM_DHCSR_DBGKEY | CORTEXM_DHCSR_C_DEBUGEN);
```
There is no need to set `DHCSR_C_HALT`.